topic-operator/src/main/java/io/strimzi/operator/topic/TopicDiff.java

```diff
@@ -225,7 +225,7 @@ public static TopicDiff diff(Topic source, Topic target) {
         Objects.requireNonNull(source.getTopicName());
         Objects.requireNonNull(target.getTopicName());
         if (!source.getTopicName().equals(target.getTopicName())) {
-            throw new IllegalArgumentException();
+            throw new IllegalArgumentException("Kafka topics cannot be renamed, but KafkaTopic's spec.topicName has changed.");
```

Comment:
I wonder if we really still need this. I would have thought that we were not detecting name changes before we get as far as doing the diff.

Reply:
IIRC this path is active in periodical reconciliation.